step.go

```diff
@@ -400,325 +363,47 @@ func exportOutput(artifacts []string) (string, string, error) {
-		if err != nil {
-			return "", fmt.Errorf("failed to fetch project's targets, error: %s", err)
-		}
-
-		buildSettings, err = proj.TargetBuildSettings(mainTarget.Name, configuration, customOptions...)
-		if err != nil {
-			return "", fmt.Errorf("failed to parse project (%s) build settings, error: %s", projectPath, err)
-		}
-	} else if xcworkspace.IsWorkspace(projectPath) {
-		workspace, err := xcworkspace.Open(projectPath)
-		if err != nil {
-			return "", fmt.Errorf("Failed to open xcworkspace (%s), error: %s", projectPath, err)
-		}
-
-		buildSettings, err = workspace.SchemeBuildSettings(scheme.Name, configuration, customOptions...)
-		if err != nil {
-			return "", fmt.Errorf("failed to parse workspace (%s) build settings, error: %s", projectPath, err)
-		}
-	} else {
-		return "", fmt.Errorf("project file extension should be .xcodeproj or .xcworkspace, but got: %s", filepath.Ext(projectPath))
-
-	}
-
-	schemeBuildDir, err := buildSettings.String("TARGET_BUILD_DIR")
-
-	if err != nil {
-		return "", fmt.Errorf("failed to parse build settings, error: %s", err)
-	}
-
-	return schemeBuildDir, nil
-}
-
-func wrapperNameForScheme(proj xcodeproj.XcodeProj, projectPath string, scheme xcscheme.Scheme, configuration string, customOptions ...string) (string, error) {
-	// Fetch project's main target from .xcodeproject
-	var buildSettings serialized.Object
-	if xcodeproj.IsXcodeProj(projectPath) {
-		mainTarget, err := mainTargetOfScheme(scheme, proj.Proj.Targets)
-		if err != nil {
-			return "", fmt.Errorf("failed to fetch project's targets, error: %s", err)
-		}
-
-		buildSettings, err = proj.TargetBuildSettings(mainTarget.Name, configuration, customOptions...)
-		if err != nil {
-			return "", fmt.Errorf("failed to parse project (%s) build settings, error: %s", projectPath, err)
-		}
-	} else if xcworkspace.IsWorkspace(projectPath) {
-		workspace, err := xcworkspace.Open(projectPath)
-		if err != nil {
-			return "", fmt.Errorf("failed to open xcworkspace (%s), error: %s", projectPath, err)
-		}
-
-		buildSettings, err = workspace.SchemeBuildSettings(scheme.Name, configuration, customOptions...)
-		if err != nil {
-			return "", fmt.Errorf("failed to parse workspace (%s) build settings, error: %s", projectPath, err)
-		}
-	} else {
-		return "", fmt.Errorf("project file extension should be .xcodeproj or .xcworkspace, but got: %s", filepath.Ext(projectPath))
-
-	}
-
-	wrapperName, err := buildSettings.String("WRAPPER_NAME")
-
-	if err != nil {
-		return "", fmt.Errorf("failed to parse build settings, error: %s", err)
-	}
-
-	return wrapperName, nil
-}
-
 // exportArtifacts exports the main target and it's .app dependencies.
-func exportArtifacts(proj xcodeproj.XcodeProj, scheme xcscheme.Scheme, schemeBuildDir, configuration string, simulatorSDK simulatorSDK, deployDir string, customOptions ...string) ([]string, error) {
+func exportArtifacts(archivePath string, deployDir string) ([]string, error) {
```

Comment:
We have an archive path here, we could just export the (zipped) archive. This is needed as currently the step does NOT zip the exported app dir, that makes it unusable without custom script step; as deploy-to-bitrise does NOT upload unzipped dirs.